servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcRouter.java

```diff
@@ -402,16 +401,16 @@ public Completable closeAsyncGracefully() {
                         @Override
                         public Publisher<Resp> handle(final GrpcServiceContext ctx, final Publisher<Req> request) {
                             return request.firstOrError()
-                                    .recoverWith(t -> {
+                                    .onErrorMap(t -> {
                                         if (t instanceof NoSuchElementException) {
-                                            return failed(new GrpcStatus(INVALID_ARGUMENT, null,
+                                            return new GrpcStatus(INVALID_ARGUMENT, null,
```

Comment:
Worth it to initialize the cause?

Reply:
yah good point, not sure if there is a reason for it to be `null`. @idelpivnitskiy - wdyt?